share/server/views.js

```diff
@@ -36,20 +36,15 @@ var Views = (function() {
     var reduce_line = JSON.stringify(reductions);
     var reduce_length = reduce_line.length;
     var input_length =  State.line_length - code_size
-    // TODO make reduce_limit config into a number
     if (State.query_config && State.query_config.reduce_limit &&
-          reduce_length > 4096 && ((reduce_length * 2) > input_length)) {
+        reduce_length > State.query_config.reduce_limit_threshold &&
+        ((reduce_length * State.query_config.reduce_limit_ratio) > input_length)) {
       var log_message = [
           "Reduce output must shrink more rapidly:",
           "input size:", input_length,
           "output size:", reduce_length
       ].join(" ");
-      if (State.query_config.reduce_limit === "log") {
-          log("reduce_overflow_error: " + log_message);
-          print("[true," + reduce_line + "]");
-      } else {
-          throw(["error", "reduce_overflow_error", log_message]);
-      };
+      throw(["error", "reduce_overflow_error", log_message]);
```

Comment:
If users used `log` and node-local views this will ignore the log and break their view and start enforcing the reduce limit. Not sure the chance of those things happening: both 1) using local dbs and views and 2) setting it to `log` but it's worth pointing it out at least